packages/components/src/Actions/ActionFile/ActionFile.component.js

```diff
@@ -0,0 +1,137 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OverlayTrigger } from 'react-bootstrap';
+import classNames from 'classnames';
+import uuid from 'uuid';
+
+import TooltipTrigger from '../../TooltipTrigger';
+import CircularProgress from '../../CircularProgress';
+
+import Icon from '../../Icon';
+import theme from './ActionFile.scss';
+
+const LEFT = 'left';
+const RIGHT = 'right';
+
+function getIcon({ icon, iconTransform, inProgress }) {
+	if (inProgress) {
+		return <CircularProgress size="small" key="icon" />;
+	}
+
+	if (icon) {
+		return <Icon name={icon} transform={iconTransform} key="icon" />;
+	}
+
+	return null;
+}
+getIcon.propTypes = {
+	icon: PropTypes.string,
+	iconTransform: PropTypes.string,
+	inProgress: PropTypes.bool,
+};
+
+function getLabel({ hideLabel, label }) {
```

Comment:
I don't understand why you need a hideLabel prop. Can't the label be shown only if specified?

Reply:
it exist for every other kind actions, and actions can be overloaded trought  settings. OSS->EE